### DISCUSSION TITLE
Filters not working since switch to v5

### DISCUSSION THREAD

**Discussion by lukepower** [Laravel-Backpack/community-forum#260] [Q&A (Help)] [answered]
Hi there,

I am using Backpack since version 3 and am happy about it. However, since the upgrade to v5, I found out that the filters I added on the list's are not I do have the PRO license installed as well.

Also, it seems that the responsive tables for the listings are not really working, as it neither resizes the table nor moves elements into the clickable "plus sign" on the left...

Any ideas?

Thanks
Lukas

**Comment by ziming** [comment]
filters work for me in v5. can u post some sample code?

**Reply by lukepower** [reply]
Sure:

`  $this->crud->addColumn([
            'name' => 'id',
            'label' => "RID",
            'searchlogic' => function ($query, $column, $searchTerm) {
                $query->orWhere('id', '=', $searchTerm);
            },
        ]);



        $this->crud->addColumn([
            'name' => 'name',
            'label' => "Name",
        ]);
        $this->crud->addColumn([
            'name' => 'updated_at',
            'label' => "Updated",
        ]);
        $this->crud->addColumn([
            'name' => 'enabled',
            'label' => "Enabled",
            'type'  => 'check'
        ]);
        $this->crud->addColumn([
            'name' => 'status',
            'label' => "Status (new)",
        ]);

        $this->crud->addColumn([
            'name' => 'failedPlans',
            'label' => "Failed"
        ]);
        $this->crud->addColumn([
            'name' => 'archived',
            'label' => "Archived"
        ]);



        $this->crud->addColumn([
            'label' => 'User', // Label for HTML form field
            'type'  => 'select',  // HTML element which displaying transactions
            'name'  => 'user_id', // Table column which is FK for Customer table
            'entity'=> 'user', // Function (method) in Customer model which retur$
            'attribute' => 'name', // Column which user see in select box
            'model' => 'App\User' // Model which contain FK
        ]);


        $this->crud->addFilter([
            'type'  => 'simple',
            'name'  => 'active',
            'label' => 'exclude Archived'
            ],
            false,
            function() { // if the filter is active
                $this->crud->addClause('where', 'archived', '0');
        } );

        // dropdown filter
        $this->crud->addFilter([
            'name'  => 'status',
            'type'  => 'dropdown',
            'label' => 'Status'
        ], [
            'Completed',
            'Saved but not submitted',
            'Submitted',
            'Submission error'
        ], function($value) { // if the filter is active
            switch ($value)
            {
                case 0: $value = 'completed'; break;
                case 1: $value = 'saved_not_yet_submitted'; break;
                case 2: $value = 'submitted-to-observatory'; break;
                case 3: $value = 'archived_but_still_on_observatory'; break;
                case 4: $value = 'submitted-to-observatory-error'; break;
            }
             $this->crud->addClause('where', 'status', $value);
        });`
 First issue is that if I use the default searchbox I cannot search by "RID" (which is the model's id), this used to work well on v4.
And the other two filters won't work as well. I am not sure where I am doing it wrong, honestly. Maybe I messed up something during the upgrade?

**Reply by ziming** [reply, marked as answer]
Hmm the code look fine to me at 1st glance. But I will do a few things differently, maybe this might solve your problem.

1. Use a text filter instead of overriding searchLogic for your id column. So remove your searchLogic function and add this filter instead

```php
$this->crud->addFilter(
            [
                'type' => 'text',
                'name' => 'id',
                'label' => 'R ID',
            ],
            false,
            function ($value) {
                $this->crud->addClause('where', 'id', $value);
            }
        );
```

2. Your simple filter for excluded archived looks fine to me, but I wonder if changing '0' to false or 0 will fix things. You may want to give it a try to see if it makes a difference or not.

3.  For your dropdown filter, instead of I noticed in your switch statement some words are separated by '-' and some are separated by '_'. You may want to check your database table to see if it is indeed the case as this is not consistent.
